### PR TITLE
Removed image generation inside the training step.

### PR DESCRIPTION
**Pull request by kureta**
It was overwriting the image grid generated in `on_epoch_end`. I also made `adversarial_loss` a static method.

### CODE REVIEW

**Comment by SeanNaren** [comment]
Thanks for the PR! I think it depends if we'd like to log the image whilst training + at the epoch end as well. Any thoughts here? An alternative would be to log with two separate names like `generated_image_epoch` and `generated_image_step` or something.

**Comment by kureta** [comment]
Happy to help! I think generating a sample at every step would be overkill. On my machine, I get around 17 iterations per second. You couldn't even see most of the generated images. But there is another problem. Since `global_step` is always 0 when calling `add_image` in `training_step`, tensorboard only show's the 0th image and never updates it. I am not sure if this is expected behavior, or a bug on the tensorboard side. In this state, you can only see updated image from `on_epoch_end` for a fraction of a second before it gets overwritten by the 0th `training_step` sample.

On a side note, here is no reason for `generated_images` in this section to be an instance variable.

**Comment by SeanNaren** [comment]
Understood, I think it makes sense then to remove the per train step image and stick with just the final epoch image then!

**Review by SeanNaren — APPROVED** [review]

**Comment by kureta** [comment]
There are a number of other small problems with this example. An unused import `OrderedDict`, use of `nn.Sigmoid`, which causes an error when trying to use 16-bit precision, vs no sigmoid and `binary_cross_entropy_with_logits`. Hyperparameters were not set in accordance with the best practices in the documentation. There is no separation between program level args, model args, and training args and it is impossible to set training flags from command line (such as `--precision=16`).

I have fixed all of these. Should I create a new branch and a new pull request, or commit my changes to this branch? Sorry, I don't have a lot of experience contributing to an open-source project.

**Comment by SeanNaren** [comment]
Make a separate PR for this, always better to make smaller PRs that fix things over time to reduce bugs appearing!

**Comment by kureta** [comment]
This fix was included in #4233 and it is now merged.
